franklab_mountainsort/core.py

```diff
@@ -313,19 +314,18 @@ def get_mda_files_dataframe(data_path, recursive=False):
             .join(geom_df)
             .replace(dict(geom_filepath={np.nan: None})))
 
-
 def _get_mda_file_information(mda_file):
+    mda_re = re.compile(
+        "^(?:(\d*)_)(?:(\w*)_)(\d*)(?:_(\w*)){0,1}(?:\.[a-zA-Z]*(\d*)){0,1}\.\w*$"
+    )
     try:
-        date, animal, epoch, other = os.path.basename(mda_file).split('_')
-        date, epoch = int(date), int(epoch)
-        task, electrode_name, _ = other.split('.')
-        electrode_number = int(electrode_name.strip('nt'))
-
+        match_re = mda_re.match(os.path.basename(mda_file))
+        date, animal, epoch, task, electrode_number = match_re.groups()
+        date, epoch, electrode_number = int(date), int(epoch), int(electrode_number)
```

Comment:
This fails when the electrode_number is None because it None can't be cast to an int. But is that the case often?

Reply:
Also how often is it the case that files don't have the .nt? Do your files have it?

Reply:
thanks for the quick turn around @edeno. you are right that mda files should always have ".nt" I will change this to make it required.